pkg/util/validator.go

```diff
@@ -13,8 +13,8 @@ import (
 )
 
 func ValidateSubnet(subnet kubeovnv1.Subnet) error {
-	if subnet.Spec.Gateway != "" && !CIDRContainIP(subnet.Spec.CIDRBlock, subnet.Spec.Gateway) {
-		return fmt.Errorf(" gateway %s is not in cidr %s", subnet.Spec.Gateway, subnet.Spec.CIDRBlock)
+	if subnet.Spec.Gateway != "" && (!CIDRContainIP(subnet.Spec.CIDRBlock, subnet.Spec.Gateway) || ValidateNetworkBroadcast(subnet.Spec.CIDRBlock, subnet.Spec.Gateway) != nil) {
+		return fmt.Errorf("gateway %s is not in cidr %s", subnet.Spec.Gateway, subnet.Spec.CIDRBlock)
```

Comment:
We should provide detailed information/log if the gateway is network/broadcast address.